src/main/cpp/src/ZOrderJni.cpp

```diff
@@ -35,4 +35,20 @@ Java_com_nvidia_spark_rapids_jni_ZOrder_interleaveBits(JNIEnv *env, jclass, jlon
   CATCH_STD(env, 0);
 }
 
+JNIEXPORT jlong JNICALL
+Java_com_nvidia_spark_rapids_jni_ZOrder_hilbertIndex(JNIEnv *env, jclass, jint num_bits, jlongArray input_columns) {
```

Comment:
Why not pass in a table from Java instead?

Reply:
It is minor but this reduces the number of JNI calls and in my opinion fits better with how Spark uses it. If we pass a table down from java then I have to create a table in java, which will require a JNI call.  I use the table internally to verify that the length of the columns are all the same, and so I can reuse the existing device_table_view.  Why does java need to know that is how I am doing it.